app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java

```diff
@@ -1,31 +1,184 @@
 import com.nervousfish.nervousfish.service_locator.IServiceLocatorCreator;
 import com.nervousfish.nervousfish.service_locator.ModuleWrapper;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
 /**
- * An adapter to the Gson database library
+ * An adapter to the GSON database library
  */
 public final class GsonDatabaseAdapter implements IDatabase {
 
+    private final static Type TYPE_CONTACT_LIST = new TypeToken<ArrayList<Contact>>(){}.getType();
+    private final IConstants constants;
+    private final GsonBuilder gsonBuilder = new GsonBuilder()
+            .registerTypeHierarchyAdapter(IKey.class, new KeyAdapter());
+    private final Gson gsonParser = this.gsonBuilder.create();
+
     /**
      * Prevents construction from outside the class.
+     *
      * @param serviceLocatorCreator The object responsible for creating the service locator
      */
-    @SuppressWarnings("PMD.UnusedFormalParameter")
     private GsonDatabaseAdapter(final IServiceLocatorCreator serviceLocatorCreator) {
-        // final IServiceLocator serviceLocator = serviceLocatorCreator.getServiceLocator();
+        final IServiceLocator serviceLocator = serviceLocatorCreator.getServiceLocator();
+        this.constants = serviceLocator.getConstants();
+
+        try {
+            this.initializeDatabase();
+        } catch (final IOException e) {
+            // TODO: Handle failure
+            e.printStackTrace();
+        }
     }
 
     /**
-     * Creates a new instance of itself and wraps it in a {@link ModuleWrapper} so that only an {@link IServiceLocatorCreator}
-     * can access the new module to create the new {@link IServiceLocator}.
+     * Creates a new instance of itself and wraps it in a {@link ModuleWrapper} so that only an
+     * {@link IServiceLocatorCreator} can access the new module to create the new
+     * {@link IServiceLocator}.
      *
      * @param serviceLocatorCreator The service locator bridge that creates the new service locator
      * @return A wrapper around a newly created instance of this class
      */
     public static ModuleWrapper<GsonDatabaseAdapter> newInstance(final IServiceLocatorCreator serviceLocatorCreator) {
         return new ModuleWrapper<>(new GsonDatabaseAdapter(serviceLocatorCreator));
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void addContact(final Contact contact) throws IOException {
+        final String contactsPath = this.constants.getDatabaseContactsPath();
+
+        // Get the list of contacts and add the new contact
+        final List<Contact> contacts = this.getAllContacts();
```

Comment:
It's ok for this application because performance is irrelevant, but it would be more efficient to maintain an internal list of all contacts instead of getting the whole list from the database every time